plugins/plugin_utils/connectivity_utils.py

```diff
@@ -0,0 +1,373 @@
+
+                if member.name.startswith('sshproxy'):
+                    member.name = proxy_name
+                elif member.name.lower() not in ['license.txt', 'thirdpartynotice.txt']:
+                    tar.close()
+                    file_utils.delete_file(
+                        proxy_package_path,
+                        f"Failed to delete {proxy_package_path}. Please delete it manually.",
+                        True)
+                    raise AnsibleError(
+                        "Attempt to download Arc Connectivity Proxy returned unnexpected result. Please try again.")
+
+                members.append(member)
+
+        tar.extractall(members=members, path=install_dir)
+
+
+def _check_proxy_installation(install_dir, proxy_name):
+    proxy_filepath = os.path.join(install_dir, proxy_name)
+    if not os.path.isfile(proxy_filepath):
+        raise AnsibleError(
+            "Failed to install required SSH Arc Connectivity Proxy. "
+            f"Couldn't find expected file {proxy_filepath}. Please try again.")
+
+
+def _get_proxy_filename(operating_system, architecture):
+    if operating_system.lower() == 'darwin' and architecture == '386':
+        raise AnsibleError("Unsupported Darwin OS with 386 architecture.")
+    proxy_filename = \
+        f"sshProxy_{operating_system.lower()}_{architecture}_{consts.CLIENT_PROXY_VERSION.replace('.', '_')}"
+    if operating_system.lower() == 'windows':
+        proxy_filename += '.exe'
+    return proxy_filename
+
+
+def _get_older_version_proxy_path(install_dir, operating_system, architecture):
+    proxy_name = f"sshProxy_{operating_system.lower()}_{architecture}_*"
+    return os.path.join(install_dir, proxy_name)
+
+
+def _get_proxy_install_dir(arc_proxy_folder):
+    if not arc_proxy_folder:
+        return os.path.expanduser(os.path.join('~', ".clientsshproxy"))
+    return arc_proxy_folder
+
+
+def _get_client_architeture():
+    import platform
+    machine = platform.machine()
+    architecture = None
+
+    if "arm64" in machine.lower() or "aarch64" in machine.lower():
+        architecture = 'arm64'
+    elif machine.endswith('64'):
+        architecture = 'amd64'
+    elif machine.endswith('86'):
+        architecture = '386'
+    elif machine == '':
+        raise AnsibleError("Couldn't identify the platform architecture.")
+    else:
+        raise AnsibleError(f"Unsuported architecture: {machine} is not currently supported")
+
+    return architecture
+
+
+def _get_client_operating_system():
+    import platform
+    operating_system = platform.system()
+
+    if operating_system.lower() not in ('linux', 'darwin', 'windows'):
+        raise AnsibleError(f"Unsuported OS: {operating_system} platform is not currently supported")
+    return operating_system
+
+def resource(client,
+             url,
+             method,
+             body=None,
+             query_parameters={},
+             header_parameters={},
+             status_code = [200, 201, 202]):
```

Comment:
unexpected spaces around keyword / parameter equals
```suggestion
             status_code=[200, 201, 202]):
```